plugins/lang-groovy/src/test/java/org/elasticsearch/messy/tests/ExtendedStatsTests.java

```diff
@@ -502,7 +502,7 @@ private void assertShardExecutionState(SearchResponse response, int expectedFail
         ShardSearchFailure[] failures = response.getShardFailures();
         if (failures.length != expectedFailures) {
             for (ShardSearchFailure failure : failures) {
-                logger.error("Shard Failure: {}", failure.reason(), failure.toString());
+                logger.error("Shard Failure: reason [{}], failure [{}]", failure.getCause(), failure.reason(), failure.toString());
```

Comment:
We don't need to call `toString` on failure - the logger will do that for us.

It feels like failure's toString should include the reason so it'd be enough to just do
```java
logger.error("Shard Failure: [{}]", failure.getCause(), failure);
```